src/fileio.c

```diff
@@ -4872,7 +4887,7 @@ delete_recursive(char_u *name)
 	exp = vim_strsave(name);
 	if (exp == NULL)
 	    return -1;
-	if (readdir_core(&ga, exp, FALSE, NULL, NULL) == OK)
+	if (readdir_core(&ga, exp, FALSE, NULL, NULL, 0) == OK)
```

Comment:
```suggestion
	if (readdir_core(&ga, exp, FALSE, NULL, NULL, READDIR_SORT_NONE) == OK)
```
Actually, this changes the current behavior, but that may not be an issue.

Reply:
Indeed it actually changes behaviour, but then I thought, why should we actually sort the result, if we are going to delete it? And in addition it will be a small performance gain.

Reply:
I used the `0`, because READDIR_SORT_NONE is undefined for tiny builts. I now define only `READDIR_SORT_NONE` in all cases, the other ifdefs only for EVAL feature. Not pretty.

Reply:
Ah, understood.
Then, how about always defining all of them?